Microsoft.NET.Build.Containers/AuthHandshakeMessageHandler.cs

```diff
@@ -18,22 +18,23 @@ namespace Microsoft.NET.Build.Containers;
 /// </summary>
 public partial class AuthHandshakeMessageHandler : DelegatingHandler
 {
-    private record AuthInfo(Uri Realm, string Service, string Scope);
+    private record AuthInfo(Uri Realm, string Service, string? Scope);
 
     /// <summary>
-    /// Cache of most-recently-recieved token for each server.
+    /// Cache of most-recently-received token for each server.
     /// </summary>
-    private static ConcurrentDictionary<string, string> TokenCache = new();
+    private static Dictionary<string, AuthenticationHeaderValue> HostAuthenticationCache = new();
```

Comment:
Why doesn't this need to be concurrent any more? Aren't we still going to have different competing entries with e.g. different scopes?

Reply:
probably a merge issue on my end - it definitely should still be concurrent. I'll fix ASAP.

Reply:
Fixed in the latest version.